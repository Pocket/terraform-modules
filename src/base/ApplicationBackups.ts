import { Construct } from 'constructs';
import { TerraformMetaArguments } from 'cdktf';
import {
  BackupPlanRule,
  BackupPlan,
} from '@cdktf/provider-aws/lib/backup-plan';
import {
  BackupSelection,
  BackupSelectionSelectionTag,
} from '@cdktf/provider-aws/lib/backup-selection';
import { BackupVault } from '@cdktf/provider-aws/lib/backup-vault';
import { BackupVaultPolicy } from '@cdktf/provider-aws/lib/backup-vault-policy';

export interface ApplicationBackupProps extends TerraformMetaArguments {
  name: string;
  kmsKeyArn: string;
  prefix: string;
  accountId: string;
  vaultPolicy: string;
  backupPlans: {
    name: string;
    resources: string[];
    rules: Omit<BackupPlanRule, 'targetVaultName'>[];
    selectionTag: BackupSelectionSelectionTag[];
  }[];
  tags?: { [key: string]: string };
}

export class ApplicationBackup extends Construct {
  public backupPlan: BackupPlan;
  public backupSelection: BackupSelection;
  public backupPlanRule: BackupPlanRule;
  private static vault: BackupVault;

  constructor(
    scope: Construct,
    name: string,
    private config: ApplicationBackupProps,
  ) {
    super(scope, name);

    const vault = new BackupVault(this, 'backup-vault', {
      name: `${config.prefix}-${config.name}`,
      kmsKeyArn: config.kmsKeyArn,
      tags: config.tags,
      provider: config.provider,
    });

    new BackupVaultPolicy(this, 'backup-vault-policy', {
      backupVaultName: vault.name,
      policy: config.vaultPolicy,
      provider: config.provider,
    });

    config.backupPlans.forEach((plan) => {
      const backupPlan = new BackupPlan(this, 'backup-plan', {
        name: plan.name,
        rule: plan.rules.map((rule) => ({
          ...rule,
          targetVaultName: vault.name,
        })),
        tags: config.tags,
        provider: config.provider,
      });

      new BackupSelection(this, 'backup-selection', {
        name: `${config.prefix}-Backup-Selection`,
        planId: backupPlan.id,
        iamRoleArn: `arn:aws:iam::${config.accountId}:role/service-role/AWSBackupDefaultServiceRole`,
        resources: plan.resources,
        selectionTag: plan.selectionTag,
        provider: config.provider,
      });
    });
  }
}
