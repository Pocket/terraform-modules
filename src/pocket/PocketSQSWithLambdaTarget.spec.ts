import { TerraformStack, Testing } from 'cdktf';
import { LAMBDA_RUNTIMES } from '../base/ApplicationVersionedLambda';
import {
  PocketSQSWithLambdaTarget,
  PocketSQSWithLambdaTargetProps,
} from './PocketSQSWithLambdaTarget';

const config: PocketSQSWithLambdaTargetProps = {
  name: 'test-sqs-lambda',
  lambda: {
    runtime: LAMBDA_RUNTIMES.PYTHON38,
    handler: 'index.handler',
  },
};

test('renders a plain sqs queue and lambda target', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  new PocketSQSWithLambdaTarget(stack, 'test-sqs-lambda', {
    ...config,
  });

  expect(Testing.synth(stack)).toMatchSnapshot();
});

test('renders a plain sqs queue with a deadletter and lambda target', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  new PocketSQSWithLambdaTarget(stack, 'test-sqs-lambda', {
    ...config,
    sqsQueue: {
      maxReceiveCount: 3,
    },
  });

  expect(Testing.synth(stack)).toMatchSnapshot();
});

test('renders a plain sqs queue with a deadletter and lambda target', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  new PocketSQSWithLambdaTarget(stack, 'test-sqs-lambda', {
    ...config,
    sqsQueue: {
      maxReceiveCount: 3,
    },
  });

  expect(Testing.synth(stack)).toMatchSnapshot();
});
