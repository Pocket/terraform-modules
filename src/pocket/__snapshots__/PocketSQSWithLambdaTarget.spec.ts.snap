// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a plain sqs queue and lambda target 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_s3_bucket\\": {
      \\"test-sqs-lambda_code-bucket_34D549A2\\": {
        \\"acl\\": \\"private\\",
        \\"bucket\\": \\"pocket-test-sqs-lambda\\",
        \\"force_destroy\\": true,
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/code-bucket\\",
            \\"uniqueId\\": \\"test-sqs-lambda_code-bucket_34D549A2\\"
          }
        }
      }
    },
    \\"aws_s3_bucket_public_access_block\\": {
      \\"test-sqs-lambda_code-bucket-public-access-block_942034EC\\": {
        \\"block_public_acls\\": true,
        \\"block_public_policy\\": true,
        \\"bucket\\": \\"\${aws_s3_bucket.test-sqs-lambda_code-bucket_34D549A2.id}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/code-bucket-public-access-block\\",
            \\"uniqueId\\": \\"test-sqs-lambda_code-bucket-public-access-block_942034EC\\"
          }
        }
      }
    },
    \\"aws_iam_role\\": {
      \\"test-sqs-lambda_execution-role_7323CCD9\\": {
        \\"assume_role_policy\\": \\"\${data.aws_iam_policy_document.test-sqs-lambda_assume-policy-document_98DE5C4D.json}\\",
        \\"name\\": \\"test-sqs-lambda-ExecutionRole\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/execution-role\\",
            \\"uniqueId\\": \\"test-sqs-lambda_execution-role_7323CCD9\\"
          }
        }
      }
    },
    \\"aws_iam_role_policy\\": {
      \\"test-sqs-lambda_execution-role-policy_D44139EE\\": {
        \\"name\\": \\"test-sqs-lambda-ExecutionRolePolicy\\",
        \\"policy\\": \\"\${data.aws_iam_policy_document.test-sqs-lambda_execution-policy-document_2C5F0106.json}\\",
        \\"role\\": \\"\${aws_iam_role.test-sqs-lambda_execution-role_7323CCD9.name}\\",
        \\"depends_on\\": [
          \\"aws_iam_role.test-sqs-lambda_execution-role_7323CCD9\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/execution-role-policy\\",
            \\"uniqueId\\": \\"test-sqs-lambda_execution-role-policy_D44139EE\\"
          }
        }
      }
    },
    \\"aws_lambda_function\\": {
      \\"test-sqs-lambda_C2DB9DC9\\": {
        \\"filename\\": \\"\${data.archive_file.test-sqs-lambda_lambda-default-file_52E1CC8A.output_path}\\",
        \\"function_name\\": \\"test-sqs-lambda-Function\\",
        \\"handler\\": \\"index.handler\\",
        \\"publish\\": true,
        \\"role\\": \\"\${aws_iam_role.test-sqs-lambda_execution-role_7323CCD9.arn}\\",
        \\"runtime\\": \\"python3.8\\",
        \\"source_code_hash\\": \\"\${data.archive_file.test-sqs-lambda_lambda-default-file_52E1CC8A.output_base64sha256}\\",
        \\"timeout\\": 5,
        \\"vpc_config\\": [],
        \\"lifecycle\\": {
          \\"ignore_changes\\": [
            \\"filename\\",
            \\"source_code_hash\\"
          ]
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/lambda\\",
            \\"uniqueId\\": \\"test-sqs-lambda_C2DB9DC9\\"
          }
        }
      }
    },
    \\"aws_cloudwatch_log_group\\": {
      \\"test-sqs-lambda_log-group_F69AB78F\\": {
        \\"name\\": \\"/aws/lambda/\${aws_lambda_function.test-sqs-lambda_C2DB9DC9.function_name}\\",
        \\"retention_in_days\\": 14,
        \\"depends_on\\": [
          \\"aws_lambda_function.test-sqs-lambda_C2DB9DC9\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/log-group\\",
            \\"uniqueId\\": \\"test-sqs-lambda_log-group_F69AB78F\\"
          }
        }
      }
    },
    \\"aws_lambda_alias\\": {
      \\"test-sqs-lambda_alias_2C2A09A2\\": {
        \\"function_name\\": \\"\${aws_lambda_function.test-sqs-lambda_C2DB9DC9.function_name}\\",
        \\"function_version\\": \\"\${split(\\\\\\":\\\\\\", aws_lambda_function.test-sqs-lambda_C2DB9DC9.qualified_arn)[7]}\\",
        \\"name\\": \\"DEPLOYED\\",
        \\"depends_on\\": [
          \\"aws_lambda_function.test-sqs-lambda_C2DB9DC9\\"
        ],
        \\"lifecycle\\": {
          \\"ignore_changes\\": [
            \\"function_version\\"
          ]
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/alias\\",
            \\"uniqueId\\": \\"test-sqs-lambda_alias_2C2A09A2\\"
          }
        }
      }
    },
    \\"aws_sqs_queue\\": {
      \\"test-sqs-lambda_lambda_sqs_queue_7C6D24B3\\": {
        \\"fifo_queue\\": false,
        \\"name\\": \\"test-sqs-lambda-Queue\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda_sqs_queue/sqs_queue\\",
            \\"uniqueId\\": \\"test-sqs-lambda_lambda_sqs_queue_7C6D24B3\\"
          }
        }
      }
    },
    \\"aws_lambda_event_source_mapping\\": {
      \\"test-sqs-lambda_lambda_event_source_mapping_C60D5FF2\\": {
        \\"event_source_arn\\": \\"\${aws_sqs_queue.test-sqs-lambda_lambda_sqs_queue_7C6D24B3.arn}\\",
        \\"function_name\\": \\"\${aws_lambda_alias.test-sqs-lambda_alias_2C2A09A2.arn}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda_event_source_mapping\\",
            \\"uniqueId\\": \\"test-sqs-lambda_lambda_event_source_mapping_C60D5FF2\\"
          }
        }
      }
    }
  },
  \\"data\\": {
    \\"aws_iam_policy_document\\": {
      \\"test-sqs-lambda_assume-policy-document_98DE5C4D\\": {
        \\"version\\": \\"2012-10-17\\",
        \\"statement\\": [
          {
            \\"actions\\": [
              \\"sts:AssumeRole\\"
            ],
            \\"effect\\": \\"Allow\\",
            \\"principals\\": [
              {
                \\"identifiers\\": [
                  \\"lambda.amazonaws.com\\"
                ],
                \\"type\\": \\"Service\\"
              }
            ]
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/assume-policy-document\\",
            \\"uniqueId\\": \\"test-sqs-lambda_assume-policy-document_98DE5C4D\\"
          }
        }
      },
      \\"test-sqs-lambda_execution-policy-document_2C5F0106\\": {
        \\"version\\": \\"2012-10-17\\",
        \\"statement\\": [
          {
            \\"actions\\": [
              \\"logs:CreateLogGroup\\",
              \\"logs:CreateLogStream\\",
              \\"logs:PutLogEvents\\",
              \\"logs:DescribeLogStreams\\"
            ],
            \\"effect\\": \\"Allow\\",
            \\"resources\\": [
              \\"arn:aws:logs:*:*:*\\"
            ]
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/execution-policy-document\\",
            \\"uniqueId\\": \\"test-sqs-lambda_execution-policy-document_2C5F0106\\"
          }
        }
      }
    },
    \\"archive_file\\": {
      \\"test-sqs-lambda_lambda-default-file_52E1CC8A\\": {
        \\"output_path\\": \\"index.py.zip\\",
        \\"type\\": \\"zip\\",
        \\"source\\": [
          {
            \\"content\\": \\"handler(event, context):\\\\n\\\\t print(event)\\",
            \\"filename\\": \\"index.py\\"
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/lambda-default-file\\",
            \\"uniqueId\\": \\"test-sqs-lambda_lambda-default-file_52E1CC8A\\"
          }
        }
      }
    }
  }
}"
`;

exports[`renders a plain sqs queue with a deadletter and lambda target 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_s3_bucket\\": {
      \\"test-sqs-lambda_code-bucket_34D549A2\\": {
        \\"acl\\": \\"private\\",
        \\"bucket\\": \\"pocket-test-sqs-lambda\\",
        \\"force_destroy\\": true,
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/code-bucket\\",
            \\"uniqueId\\": \\"test-sqs-lambda_code-bucket_34D549A2\\"
          }
        }
      }
    },
    \\"aws_s3_bucket_public_access_block\\": {
      \\"test-sqs-lambda_code-bucket-public-access-block_942034EC\\": {
        \\"block_public_acls\\": true,
        \\"block_public_policy\\": true,
        \\"bucket\\": \\"\${aws_s3_bucket.test-sqs-lambda_code-bucket_34D549A2.id}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/code-bucket-public-access-block\\",
            \\"uniqueId\\": \\"test-sqs-lambda_code-bucket-public-access-block_942034EC\\"
          }
        }
      }
    },
    \\"aws_iam_role\\": {
      \\"test-sqs-lambda_execution-role_7323CCD9\\": {
        \\"assume_role_policy\\": \\"\${data.aws_iam_policy_document.test-sqs-lambda_assume-policy-document_98DE5C4D.json}\\",
        \\"name\\": \\"test-sqs-lambda-ExecutionRole\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/execution-role\\",
            \\"uniqueId\\": \\"test-sqs-lambda_execution-role_7323CCD9\\"
          }
        }
      }
    },
    \\"aws_iam_role_policy\\": {
      \\"test-sqs-lambda_execution-role-policy_D44139EE\\": {
        \\"name\\": \\"test-sqs-lambda-ExecutionRolePolicy\\",
        \\"policy\\": \\"\${data.aws_iam_policy_document.test-sqs-lambda_execution-policy-document_2C5F0106.json}\\",
        \\"role\\": \\"\${aws_iam_role.test-sqs-lambda_execution-role_7323CCD9.name}\\",
        \\"depends_on\\": [
          \\"aws_iam_role.test-sqs-lambda_execution-role_7323CCD9\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/execution-role-policy\\",
            \\"uniqueId\\": \\"test-sqs-lambda_execution-role-policy_D44139EE\\"
          }
        }
      }
    },
    \\"aws_lambda_function\\": {
      \\"test-sqs-lambda_C2DB9DC9\\": {
        \\"filename\\": \\"\${data.archive_file.test-sqs-lambda_lambda-default-file_52E1CC8A.output_path}\\",
        \\"function_name\\": \\"test-sqs-lambda-Function\\",
        \\"handler\\": \\"index.handler\\",
        \\"publish\\": true,
        \\"role\\": \\"\${aws_iam_role.test-sqs-lambda_execution-role_7323CCD9.arn}\\",
        \\"runtime\\": \\"python3.8\\",
        \\"source_code_hash\\": \\"\${data.archive_file.test-sqs-lambda_lambda-default-file_52E1CC8A.output_base64sha256}\\",
        \\"timeout\\": 5,
        \\"vpc_config\\": [],
        \\"lifecycle\\": {
          \\"ignore_changes\\": [
            \\"filename\\",
            \\"source_code_hash\\"
          ]
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/lambda\\",
            \\"uniqueId\\": \\"test-sqs-lambda_C2DB9DC9\\"
          }
        }
      }
    },
    \\"aws_cloudwatch_log_group\\": {
      \\"test-sqs-lambda_log-group_F69AB78F\\": {
        \\"name\\": \\"/aws/lambda/\${aws_lambda_function.test-sqs-lambda_C2DB9DC9.function_name}\\",
        \\"retention_in_days\\": 14,
        \\"depends_on\\": [
          \\"aws_lambda_function.test-sqs-lambda_C2DB9DC9\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/log-group\\",
            \\"uniqueId\\": \\"test-sqs-lambda_log-group_F69AB78F\\"
          }
        }
      }
    },
    \\"aws_lambda_alias\\": {
      \\"test-sqs-lambda_alias_2C2A09A2\\": {
        \\"function_name\\": \\"\${aws_lambda_function.test-sqs-lambda_C2DB9DC9.function_name}\\",
        \\"function_version\\": \\"\${split(\\\\\\":\\\\\\", aws_lambda_function.test-sqs-lambda_C2DB9DC9.qualified_arn)[7]}\\",
        \\"name\\": \\"DEPLOYED\\",
        \\"depends_on\\": [
          \\"aws_lambda_function.test-sqs-lambda_C2DB9DC9\\"
        ],
        \\"lifecycle\\": {
          \\"ignore_changes\\": [
            \\"function_version\\"
          ]
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/alias\\",
            \\"uniqueId\\": \\"test-sqs-lambda_alias_2C2A09A2\\"
          }
        }
      }
    },
    \\"aws_sqs_queue\\": {
      \\"test-sqs-lambda_lambda_sqs_queue_redrive_sqs_queue_EBFF9A33\\": {
        \\"fifo_queue\\": false,
        \\"name\\": \\"test-sqs-lambda-Queue-Deadletter\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda_sqs_queue/redrive_sqs_queue\\",
            \\"uniqueId\\": \\"test-sqs-lambda_lambda_sqs_queue_redrive_sqs_queue_EBFF9A33\\"
          }
        }
      },
      \\"test-sqs-lambda_lambda_sqs_queue_7C6D24B3\\": {
        \\"fifo_queue\\": false,
        \\"name\\": \\"test-sqs-lambda-Queue\\",
        \\"redrive_policy\\": \\"{\\\\\\"maxReceiveCount\\\\\\":3,\\\\\\"deadLetterTargetArn\\\\\\":\\\\\\"\${aws_sqs_queue.test-sqs-lambda_lambda_sqs_queue_redrive_sqs_queue_EBFF9A33.arn}\\\\\\"}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda_sqs_queue/sqs_queue\\",
            \\"uniqueId\\": \\"test-sqs-lambda_lambda_sqs_queue_7C6D24B3\\"
          }
        }
      }
    },
    \\"aws_lambda_event_source_mapping\\": {
      \\"test-sqs-lambda_lambda_event_source_mapping_C60D5FF2\\": {
        \\"event_source_arn\\": \\"\${aws_sqs_queue.test-sqs-lambda_lambda_sqs_queue_7C6D24B3.arn}\\",
        \\"function_name\\": \\"\${aws_lambda_alias.test-sqs-lambda_alias_2C2A09A2.arn}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda_event_source_mapping\\",
            \\"uniqueId\\": \\"test-sqs-lambda_lambda_event_source_mapping_C60D5FF2\\"
          }
        }
      }
    }
  },
  \\"data\\": {
    \\"aws_iam_policy_document\\": {
      \\"test-sqs-lambda_assume-policy-document_98DE5C4D\\": {
        \\"version\\": \\"2012-10-17\\",
        \\"statement\\": [
          {
            \\"actions\\": [
              \\"sts:AssumeRole\\"
            ],
            \\"effect\\": \\"Allow\\",
            \\"principals\\": [
              {
                \\"identifiers\\": [
                  \\"lambda.amazonaws.com\\"
                ],
                \\"type\\": \\"Service\\"
              }
            ]
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/assume-policy-document\\",
            \\"uniqueId\\": \\"test-sqs-lambda_assume-policy-document_98DE5C4D\\"
          }
        }
      },
      \\"test-sqs-lambda_execution-policy-document_2C5F0106\\": {
        \\"version\\": \\"2012-10-17\\",
        \\"statement\\": [
          {
            \\"actions\\": [
              \\"logs:CreateLogGroup\\",
              \\"logs:CreateLogStream\\",
              \\"logs:PutLogEvents\\",
              \\"logs:DescribeLogStreams\\"
            ],
            \\"effect\\": \\"Allow\\",
            \\"resources\\": [
              \\"arn:aws:logs:*:*:*\\"
            ]
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/execution-policy-document\\",
            \\"uniqueId\\": \\"test-sqs-lambda_execution-policy-document_2C5F0106\\"
          }
        }
      }
    },
    \\"archive_file\\": {
      \\"test-sqs-lambda_lambda-default-file_52E1CC8A\\": {
        \\"output_path\\": \\"index.py.zip\\",
        \\"type\\": \\"zip\\",
        \\"source\\": [
          {
            \\"content\\": \\"handler(event, context):\\\\n\\\\t print(event)\\",
            \\"filename\\": \\"index.py\\"
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/lambda-default-file\\",
            \\"uniqueId\\": \\"test-sqs-lambda_lambda-default-file_52E1CC8A\\"
          }
        }
      }
    }
  }
}"
`;

exports[`renders a plain sqs queue with a deadletter and lambda target 2`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"resource\\": {
    \\"aws_s3_bucket\\": {
      \\"test-sqs-lambda_code-bucket_34D549A2\\": {
        \\"acl\\": \\"private\\",
        \\"bucket\\": \\"pocket-test-sqs-lambda\\",
        \\"force_destroy\\": true,
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/code-bucket\\",
            \\"uniqueId\\": \\"test-sqs-lambda_code-bucket_34D549A2\\"
          }
        }
      }
    },
    \\"aws_s3_bucket_public_access_block\\": {
      \\"test-sqs-lambda_code-bucket-public-access-block_942034EC\\": {
        \\"block_public_acls\\": true,
        \\"block_public_policy\\": true,
        \\"bucket\\": \\"\${aws_s3_bucket.test-sqs-lambda_code-bucket_34D549A2.id}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/code-bucket-public-access-block\\",
            \\"uniqueId\\": \\"test-sqs-lambda_code-bucket-public-access-block_942034EC\\"
          }
        }
      }
    },
    \\"aws_iam_role\\": {
      \\"test-sqs-lambda_execution-role_7323CCD9\\": {
        \\"assume_role_policy\\": \\"\${data.aws_iam_policy_document.test-sqs-lambda_assume-policy-document_98DE5C4D.json}\\",
        \\"name\\": \\"test-sqs-lambda-ExecutionRole\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/execution-role\\",
            \\"uniqueId\\": \\"test-sqs-lambda_execution-role_7323CCD9\\"
          }
        }
      }
    },
    \\"aws_iam_role_policy\\": {
      \\"test-sqs-lambda_execution-role-policy_D44139EE\\": {
        \\"name\\": \\"test-sqs-lambda-ExecutionRolePolicy\\",
        \\"policy\\": \\"\${data.aws_iam_policy_document.test-sqs-lambda_execution-policy-document_2C5F0106.json}\\",
        \\"role\\": \\"\${aws_iam_role.test-sqs-lambda_execution-role_7323CCD9.name}\\",
        \\"depends_on\\": [
          \\"aws_iam_role.test-sqs-lambda_execution-role_7323CCD9\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/execution-role-policy\\",
            \\"uniqueId\\": \\"test-sqs-lambda_execution-role-policy_D44139EE\\"
          }
        }
      }
    },
    \\"aws_lambda_function\\": {
      \\"test-sqs-lambda_C2DB9DC9\\": {
        \\"filename\\": \\"\${data.archive_file.test-sqs-lambda_lambda-default-file_52E1CC8A.output_path}\\",
        \\"function_name\\": \\"test-sqs-lambda-Function\\",
        \\"handler\\": \\"index.handler\\",
        \\"publish\\": true,
        \\"role\\": \\"\${aws_iam_role.test-sqs-lambda_execution-role_7323CCD9.arn}\\",
        \\"runtime\\": \\"python3.8\\",
        \\"source_code_hash\\": \\"\${data.archive_file.test-sqs-lambda_lambda-default-file_52E1CC8A.output_base64sha256}\\",
        \\"timeout\\": 5,
        \\"vpc_config\\": [],
        \\"lifecycle\\": {
          \\"ignore_changes\\": [
            \\"filename\\",
            \\"source_code_hash\\"
          ]
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/lambda\\",
            \\"uniqueId\\": \\"test-sqs-lambda_C2DB9DC9\\"
          }
        }
      }
    },
    \\"aws_cloudwatch_log_group\\": {
      \\"test-sqs-lambda_log-group_F69AB78F\\": {
        \\"name\\": \\"/aws/lambda/\${aws_lambda_function.test-sqs-lambda_C2DB9DC9.function_name}\\",
        \\"retention_in_days\\": 14,
        \\"depends_on\\": [
          \\"aws_lambda_function.test-sqs-lambda_C2DB9DC9\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/log-group\\",
            \\"uniqueId\\": \\"test-sqs-lambda_log-group_F69AB78F\\"
          }
        }
      }
    },
    \\"aws_lambda_alias\\": {
      \\"test-sqs-lambda_alias_2C2A09A2\\": {
        \\"function_name\\": \\"\${aws_lambda_function.test-sqs-lambda_C2DB9DC9.function_name}\\",
        \\"function_version\\": \\"\${split(\\\\\\":\\\\\\", aws_lambda_function.test-sqs-lambda_C2DB9DC9.qualified_arn)[7]}\\",
        \\"name\\": \\"DEPLOYED\\",
        \\"depends_on\\": [
          \\"aws_lambda_function.test-sqs-lambda_C2DB9DC9\\"
        ],
        \\"lifecycle\\": {
          \\"ignore_changes\\": [
            \\"function_version\\"
          ]
        },
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/alias\\",
            \\"uniqueId\\": \\"test-sqs-lambda_alias_2C2A09A2\\"
          }
        }
      }
    },
    \\"aws_sqs_queue\\": {
      \\"test-sqs-lambda_lambda_sqs_queue_redrive_sqs_queue_EBFF9A33\\": {
        \\"fifo_queue\\": false,
        \\"name\\": \\"test-sqs-lambda-Queue-Deadletter\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda_sqs_queue/redrive_sqs_queue\\",
            \\"uniqueId\\": \\"test-sqs-lambda_lambda_sqs_queue_redrive_sqs_queue_EBFF9A33\\"
          }
        }
      },
      \\"test-sqs-lambda_lambda_sqs_queue_7C6D24B3\\": {
        \\"fifo_queue\\": false,
        \\"name\\": \\"test-sqs-lambda-Queue\\",
        \\"redrive_policy\\": \\"{\\\\\\"maxReceiveCount\\\\\\":3,\\\\\\"deadLetterTargetArn\\\\\\":\\\\\\"\${aws_sqs_queue.test-sqs-lambda_lambda_sqs_queue_redrive_sqs_queue_EBFF9A33.arn}\\\\\\"}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda_sqs_queue/sqs_queue\\",
            \\"uniqueId\\": \\"test-sqs-lambda_lambda_sqs_queue_7C6D24B3\\"
          }
        }
      }
    },
    \\"aws_lambda_event_source_mapping\\": {
      \\"test-sqs-lambda_lambda_event_source_mapping_C60D5FF2\\": {
        \\"event_source_arn\\": \\"\${aws_sqs_queue.test-sqs-lambda_lambda_sqs_queue_7C6D24B3.arn}\\",
        \\"function_name\\": \\"\${aws_lambda_alias.test-sqs-lambda_alias_2C2A09A2.arn}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda_event_source_mapping\\",
            \\"uniqueId\\": \\"test-sqs-lambda_lambda_event_source_mapping_C60D5FF2\\"
          }
        }
      }
    }
  },
  \\"data\\": {
    \\"aws_iam_policy_document\\": {
      \\"test-sqs-lambda_assume-policy-document_98DE5C4D\\": {
        \\"version\\": \\"2012-10-17\\",
        \\"statement\\": [
          {
            \\"actions\\": [
              \\"sts:AssumeRole\\"
            ],
            \\"effect\\": \\"Allow\\",
            \\"principals\\": [
              {
                \\"identifiers\\": [
                  \\"lambda.amazonaws.com\\"
                ],
                \\"type\\": \\"Service\\"
              }
            ]
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/assume-policy-document\\",
            \\"uniqueId\\": \\"test-sqs-lambda_assume-policy-document_98DE5C4D\\"
          }
        }
      },
      \\"test-sqs-lambda_execution-policy-document_2C5F0106\\": {
        \\"version\\": \\"2012-10-17\\",
        \\"statement\\": [
          {
            \\"actions\\": [
              \\"logs:CreateLogGroup\\",
              \\"logs:CreateLogStream\\",
              \\"logs:PutLogEvents\\",
              \\"logs:DescribeLogStreams\\"
            ],
            \\"effect\\": \\"Allow\\",
            \\"resources\\": [
              \\"arn:aws:logs:*:*:*\\"
            ]
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/execution-policy-document\\",
            \\"uniqueId\\": \\"test-sqs-lambda_execution-policy-document_2C5F0106\\"
          }
        }
      }
    },
    \\"archive_file\\": {
      \\"test-sqs-lambda_lambda-default-file_52E1CC8A\\": {
        \\"output_path\\": \\"index.py.zip\\",
        \\"type\\": \\"zip\\",
        \\"source\\": [
          {
            \\"content\\": \\"handler(event, context):\\\\n\\\\t print(event)\\",
            \\"filename\\": \\"index.py\\"
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/test-sqs-lambda/lambda/lambda-default-file\\",
            \\"uniqueId\\": \\"test-sqs-lambda_lambda-default-file_52E1CC8A\\"
          }
        }
      }
    }
  }
}"
`;
